python_modules/dagster/dagster/_core/definitions/data_time.py

```diff
@@ -513,7 +532,12 @@ def get_minutes_overdue(
                 "Cannot calculate minutes late for asset without a FreshnessPolicy"
             )
 
+        if self.asset_graph.is_source(asset_key):
+            current_data_time = self._get_source_data_time(asset_key, current_time=evaluation_time)
```

Comment:
Should `get_current_data_time` be the one that delegates to `_get_source_data_time`?

Reply:
yes- just did it here to isolate it from other get_current_data_time callsites to start. This leads to the question of how data time and data version should interact since get_current_data_time recurses up to source assets to get the time- if it's set explicitly it should probably override the existing data time calculation